Stack/Opc.Ua.Core/Types/BuiltIn/NodeId.cs

```diff
@@ -1859,6 +1872,21 @@ private static void FormatIdentifier(IFormatProvider formatProvider, StringBuild
                 }
             }
         }
+
+        /// <summary>
+        /// Validate that an immutable NodeId is not overwritten.
+        /// </summary>
+        /// <exception cref="InvalidOperationException"></exception>
+        [Conditional("IMMUTABLENULLNODEID")]
+        private void ValidateImmutableNodeIdIsNotModified()
+        {
+#if IMMUTABLENULLNODEID
```

Comment:
could also be checked by comparing reference of s_null, then there is no problem with the derived class

Reply:
Ok, all these changes are done to protect this NodeId.Null singleton and are done at runtime. To be honest I do not know the background of why is these are needed and how can this help (validation being done at runtime).

Reply:
Everything looks ok from the coding point of view even if "is" is used (preferably on sealed classes) or your second idea to compare to s_null directly.